### ISSUE TITLE
Crossposts to same community with some time difference need to be not hidden

### ISSUE THREAD

**Issue by poVoq** [LemmyNet/lemmy-ui#2104]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

Continuation from here: https://github.com/LemmyNet/lemmy-ui/issues/1243

Basically when someone makes a post to the same community with the same link again (because of some discussion referencing the same thing at a later point in time for example) the newer post gets hidden by the cross-post functionality.

This is a bit unexpected behaviour as those post can have totally different contents.

One suggestion was to chose which post to hide based on votes and age, but maybe such cross-posts to the same(!) community should just not be hidden at all?

**Comment by testman42** [comment]
Maybe some two-layer UI thing, where it first lists just the communities where the post has been cross-posted to, and when the user clicks on the name of the community, the list of the relevant duplicate posts appears, sorted by time.

**Comment by SorteKanin** [comment]
>One suggestion was to chose which post to hide based on votes and age, but maybe such cross-posts to the same(!) community should just not be hidden at all?

I definitely feel that hiding crossposts in the same community is just confusing and should never have been a thing in the first place. Just show the posts regardless of the fact that they are crossposted. They are still separate posts after all and may contain entirely separate discussions.

**Comment by dessalines** [comment]
I don't understand this. The newer post is never completely hidden, its just shown in the cross_post simplifier ( and that should only be a case where its the same page of results, IE within a short time frame). 

I'm not sure any way to fix this other than just removing the dupe simplifier altogether.

**Comment by poVoq** [comment]
Back then when I opened this issue, the post was definitely hidden. Showing some link in an unrelated older post is not in any way intuitive and basically impossible to find.

Given what you wrote just now, the only thing that comes to mind would be that it might have been in a slow moving community and the older post would have been still visible on the same page.

**Comment by dessalines** [comment]
> Given what you wrote just now, the only thing that comes to mind would be that it might have been in a slow moving community and the older post would have been still visible on the same page.

Yep that's likely the case. 

> Would that be possible if the posts are in the same community?

I spose we could remove the cross-post dupe logic for communities... not sure if it'd be worth it tho. This all seems a bit overkill to me, if the same link is being posted twice in either an inactive community months apart, or an active community (which should be removed anyway as a fast duplicate), you'd probably want to centralize the discussion(plus the newer post is still available to click).

I can't think of a single good use case for this.

**Comment by Die4Ever** [comment]
As seen in https://github.com/LemmyNet/lemmy-ui/issues/2323

I've found this issue to be very confusing and unintuitive. Nobody wants this behavior, it's usually accidental and it hides discussions. If it hides the newer post then people are missing out on news/information, if it hides the older post then people are missing discussions. If you're browsing a community then you're interested in the community, this is not the All feed or anything like that.

What if the link is to a Twitch channel? That's definitely something that can be posted multiple times, like the Games Done Quick channel. Or it could be a stats page that gets updated. Or even a GitHub link like a pull request (yay some Lemmy feature just got merged). There's plenty of uses for posting the same URL again but with a different discussion.

**Comment by Die4Ever** [comment]
Also this issue is potentially dangerous for moderators but seeing a post in a community they moderate.

**Comment by Die4Ever** [comment]
The fact that it confused Lemmy DEVELOPERS and it took you time to figure out what was happening, is a good indicator that casual users will be extremely confused, find Lemmy to be unintuitive, and switch back to Reddit. 

Here is an example of how confusing it can be: https://github.com/LemmyNet/lemmy-ui/issues/1243

**Comment by dessalines** [comment]
I'm not confused, I just think you don't understand the purpose of the cross-post simplifier, which is to not show the same URL link full size multiple times in your feed. 

Again, every post is still accessible by clicking its smaller link, nothing is completely hidden. I want this behavior, and I'm sure most other people don't want to see full size post listings for 10 of the exact same cross posts to different communities.

#1243 was a different bug I fixed that showed the same post as a cross-post.

**Comment by Die4Ever** [comment]
> full size post listings for 10 of the exact same cross posts to different communities

This bug report is only about viewing a single community page, no different communities. Quoting the bug report that spawned this one:

> Somehow it ended up being cross-posted to the same community and if you go to https://slrpnk.net/c/meta it is hidden.

My bug report title states that more explicitly but you closed that as a duplicate of this. (if I'm wrong about the intent of this bug report, then my bug report is actually not a duplicate and should maybe be reopened)

So this is about the community feed, not Subscribed or Local or All.

> Again, every post is still accessible by clicking its smaller link, nothing is completely hidden.

But if I want to see new posts in a community I have to click on tons of posts to open them, check the crossposts lists, click on each of those for the same community, and then check the age of each crosspost once it's open.

**Comment by poVoq** [comment]
I think the core difference here is that in many forum like communities the URL is not the important part of the post.

It might indeed reference the same thing like a wiki page or a streaming channel, but the important part is the message that comes with it. Hiding the entire post because someone else weeks before happened to use the same URL in their post is very unintuitive and generally problematic.

**Comment by dessalines** [comment]
Okay I looked into it and this was a 1-line fix.

**Comment by Die4Ever** [comment]
> Okay I looked into it and this was a 1-line fix.

nice, any thoughts about doing this for the Moderator View too?

**Comment by SleeplessOne1917** [comment]
Adding my two cents, I find the way crossposts are handled in general to be pretty unintuitive. In addition to what others have pointed out in this thread about different posts with the same URL always being treated as crossposts (regardless of whether or not the poster intended that to be the case), it also means that crossposting a text-only post to a different community doesn't actually create a crosspost. While the body of the crosspost includes a link to the original along with the contents in block quotes, it is always possible for the user to modify the body and/or title to remove any reference to the original post; at that point, the user might as well create a post normally.

**Comment by dessalines** [comment]
I'd def like to have some sort of a cross-post simplifier for non-url posts, but unfortunately it'd need to be some kind of explicit link, whereas the URL provides a common way to simplify them, and I can't think of a simple way to do it (besides maybe link the original post as the url in the cross-post, but then we'd possibly get a lot of chaining).
